adala/utils/llm.py

```diff
@@ -0,0 +1,235 @@
+import asyncio
+import instructor
+import litellm
+import traceback
+import multiprocessing as mp
+from typing import Optional, Dict, Any, List, Type, Union
+from pydantic import BaseModel, Field
+
+instructor_client = instructor.from_litellm(litellm.completion)
+async_instructor_client = instructor.from_litellm(litellm.acompletion)
+
+
+class LLMResponse(BaseModel):
+    data: Dict = Field(default_factory=dict)
+    adala_error: bool = Field(default=False, serialization_alias='_adala_error')
+    adala_message: str = Field(default=None, serialization_alias='_adala_message')
+    adala_details: str = Field(default=None, serialization_alias='_adala_details')
+
+
+def get_messages(user_prompt: str, system_prompt: Optional[str] = None, instruction_first: bool = True):
+    messages = [{'role': 'user', 'content': user_prompt}]
+    if system_prompt:
+        if instruction_first:
+            messages.insert(0, {'role': 'system', 'content': system_prompt})
+        else:
+            messages[0]['content'] += system_prompt
```

Comment:
would this only be used if the runtime did not support a system prompt, is there another reason why you might do this? should system_prompt be added to the beginning of the message or passed as a separate user message anyway?

Reply:
I imagine the situation where you have to instruct your LLM AFTER providing the long context, to avoid lost-in-the-middle effect. 
However, I think overall `instructions` parameter can be redundant from the UX prospective and we can let user define the structure of the prompt directly in `input_template`

Reply:
anyway, changing this behavior is out of the scope as it is how all skills are currently working.